src/riscv/lib/src/jit/builder/sequence/instr_map.rs

```diff
@@ -0,0 +1,137 @@
+use std::fmt::Debug;
+
+use crate::jit::builder::instruction::LoweredInstruction;
+use crate::jit::builder::instruction::Outcome;
+use crate::machine_state::memory::Address;
+use crate::parser::instruction::InstrWidth;
+
+/// The basic metadata needed for analysis that is directly extractable
+/// from the LoweredInstruction.
+#[derive(Debug, Clone)]
+pub struct InstrInfo {
+    width: InstrWidth,
+    outcomes: Vec<Outcome>,
+}
+
+impl InstrInfo {
+    pub fn width(&self) -> InstrWidth {
+        self.width
+    }
+
+    pub fn outcomes(&self) -> &[Outcome] {
+        &self.outcomes
+    }
+}
+
+/// A holder of the instructions in a sequence with quick lookup via node ID or address.
+#[derive(Debug, Clone)]
+pub struct InstrMap<V> {
+    addresses: Vec<Key>,
+    instructions: Vec<V>,
+}
+
+type Key = Address;
+
+#[derive(Debug, Clone, Copy)]
+pub(super) struct NodeID(usize);
```

Comment:
```suggestion
pub(super) struct InstrId(usize);
```

In the JIT we work with instructions, let's not introduce a new concept of nodes unless it is really needed. Instruction identifier can be understood without much context.

Reply:
fair enough - done [here](https://github.com/tezos/riscv-pvm/compare/5381a384a59497445686c2eac9da457c681829e5..08b2f4b)